### ISSUE TITLE
[FR] Virtual parts should be able to be included in Sals Order.

### ISSUE THREAD

**Issue by mylesagray** [inventree/InvenTree#7499]
### Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find a similar feature request

### Problem statement

Use case: We have a workshop, we have all consumables (oils, filters, coolants, etc) as well as parts (which also end up fitted or consumed on vehicles) in Inventree, we would like to create "services" like "Engine Oil change", "Transmission Service", "Filter refresh".

We achieved this (we thought) using virtual parts, by creating these "services" as virtual parts, and included a standardised BOM that included for example Oil of a certain amount, a filter, and an O-ring.

We created an over-arching "Oil Change" part, that is a template and has a BOM that allows substitutions for particular oil types (some of our vehicles take specialised high performance oils, others don't) or filters.

We then create other virtual parts from that "prototype" part, for say "Audi RS3 Oil Change", or "Lotus Exige Oil Change" which contain the BOM for the specific oils and filters for those vehicles this allows us to have an easy standard BOM for these services.

The problem occurs when we want to create an SO for a few services and then have the stock for the consumed items reduced by the amounts in the BOM - because the parts are virtual (we aren't "creating" a new item - it's consumed in the vehicle, so there is no output "part") we can't add them to the SO, and we can't use the BOM feature.

### Suggested solution

Virtual parts should be allowed to be added to an SO.

This I imagine also rings true for software licenses, as they're not an "output" part, but they are certainly sold.

### Describe alternatives you've considered

As above in problem statement. Creating output parts when there is no tangible output part feels hacky.

Happy to sponsor this work, if needed.

**Comment by SchrodingersGat** [comment]
@mylesagray makes sense, but would require some consideration around how to "allocate" these items (which are virtual and do not have physical stock items). I'll mark this as "horizon" for now.

**Comment by fjpanag** [comment]
I would like to add another use case for salable virtual parts.

We sell some hardware controllers, which are accompanied with some software licenses, programming services, installation services etc.

These "virtual parts" will be sold, and there will be a price, but they are not actual/physical parts. Just a license for example, that you enable in a server.

Stock management or manufacturing for these parts is meaningless. But they are actually salable.

**Comment by SchrodingersGat** [comment]
@fjpanag this is a good use case, thanks.

**Comment by SchrodingersGat** [comment]
Ref: https://github.com/inventree/InvenTree/pull/10257
